lib/src/Uncertainty/Distribution/ConditionedDistribution.cxx

```diff
@@ -0,0 +1,591 @@
+{
+  const ConditionedDistribution* p_other = dynamic_cast<const ConditionedDistribution*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String ConditionedDistribution::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << ConditionedDistribution::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " distribution=" << distribution_
+      << " indices=" << conditioningIndices_
+      << " values=" << conditioningValues_;
+  return oss;
+}
+
+String ConditionedDistribution::__str__(const String & offset) const
+{
+  OSS oss(false);
+  oss << offset << getClassName() << "(distribution = " << distribution_ << ", indices = " << conditioningIndices_ << ", values = " << conditioningValues_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+ConditionedDistribution * ConditionedDistribution::clone() const
+{
+  return new ConditionedDistribution(*this);
+}
+
+void ConditionedDistribution::update()
+{
+  const UnsignedInteger fullDimension = distribution_.getDimension();
+  normalizationFactor_ = 1.0;
+  if (conditioningIndices_.getSize())
+    normalizationFactor_ = distribution_.getMarginal(conditioningIndices_).computePDF(conditioningValues_);
+  if (!(normalizationFactor_ > getPDFEpsilon()))
+    throw InvalidArgumentException(HERE) << "Conditioning vector PDF value is too low";
+  normalizationFactor_ = 1.0 / normalizationFactor_; // to avoid division
+  nonConditioningIndices_ = conditioningIndices_.complement(fullDimension);
+  setDimension(nonConditioningIndices_.getSize());
+  marginalConditionedDistribution_ = distribution_.getMarginal(nonConditioningIndices_);
+  setDescription(marginalConditionedDistribution_.getDescription());
+
+  // enable simplified path
+  useSimplifiedVersion_ = hasSimplifiedVersion(simplifiedVersion_);
+
+  computeRange();
+  isAlreadyComputedMean_ = false;
+  isAlreadyComputedCovariance_ = false;
+}
+
+/* Get the simplified version */
+Bool ConditionedDistribution::hasSimplifiedVersion(Distribution & simplified) const
+{
+  // Normal
+  Normal *p_normal = dynamic_cast<Normal *>(distribution_.getImplementation().get());
+  if (p_normal && (distribution_.getDimension() == 2) && (getDimension() == 1))
+  {
+    const Point mu(p_normal->getMu());
+    const CovarianceMatrix cov(p_normal->getCovariance());
+    const Scalar cyy = cov(conditioningIndices_[0], conditioningIndices_[0]);
+    const Scalar cxx = cov(nonConditioningIndices_[0], nonConditioningIndices_[0]);
+    const Scalar cxy = cov(1, 0);
+    const Scalar muy = mu[conditioningIndices_[0]];
+    const Scalar mux = mu[nonConditioningIndices_[0]];
+    const Scalar y = conditioningValues_[0];
+    Point muConditioned(getDimension());
+    CovarianceMatrix covConditioned(getDimension());
+    muConditioned[0] = mux + cxy / cyy * (y - muy);
+    covConditioned(0, 0) = cxx - cxy / cyy * cxy;
```

Comment:
What happens if this is zero? Can we create a Normal random vector in this case?

Reply:
diagonal terms cannot be null